File: HashTables/ransom-note/ransomNote.js
function ransomNote(magazine, note){
	let hashObject = {};
	magazine.split(" ").forEach(word => hashObject[word] = hashObject[word] ? hashObject[word]+1 : 1);

	let arr = note.split(" ");
	for(let i = 0; i < arr.length; i++){
		if(!hashObject[arr[i]])
			return "No";
		else
			hashObject[arr[i]]--;
	}
	return "Yes";
}

console.log(ransomNote("give me one grand today night", "give one grand today"));

File: HashTables/ransom-note/ransom_note.cpp
#include <iostream>
#include <unordered_map>

using namespace std;

int main(){
  int m, n; 
  unordered_map<string, int> mymap;
  string word;
  
  cin >> m >> n;
  for(int i = 0; i < m; i++){
    cin >> word;
    mymap[word]++;
  }
  
  for(int j = 0; j < n; j++){
    cin >> word;
    if(!mymap[word]){
      cout << "No" << endl;
      return 0;
    } else
      mymap[word]--;      
  }
  cout << "Yes" << endl;
  return 0;
}   